portal-api/src/v1/controllers/countries.controller.js

```diff
@@ -1,44 +1,122 @@
+ */
+const getCountryFromArray = (arr, code) => (arr.length ? arr.filter((country) => country.code === code)[0] : []);
 
+/**
+ * Sorts an array of countries.
+ *
+ * @param {Array} countries - The array of countries to be sorted.
+ * @returns {Array} - The sorted array of countries.
+ * @throws {Error} - If unable to sort countries.
+ */
 const sortCountries = (countries) => {
-  const countriesWithoutUK = countries.filter((country) => country.code !== 'GBR');
+  try {
+    const countriesWithoutUK = countries.filter((country) => country.code !== 'GBR');
 
-  const sortedArray = [
-    getCountryFromArray(countries, 'GBR'),
-    ...utils.sortArrayAlphabetically(countriesWithoutUK, 'name'),
-  ];
+    const sortedArray = [getCountryFromArray(countries, 'GBR'), ...utils.sortArrayAlphabetically(countriesWithoutUK, 'name')];
 
-  return sortedArray;
+    return sortedArray;
+  } catch (error) {
+    console.error('Unable to sort countries %o', error);
+    throw new Error('Unable to sort countries', { cause: error });
+  }
 };
 
-const findOneCountry = async (code) => {
-  const response = await externalApi.countries.getCountry(code);
-  return response;
+/**
+ * Retrieves country information from an external API based on the provided country code.
+ *
+ * @param {string} code - The country code.
+ * @returns {Promise<Object>} - A promise that resolves to the country information.
+ * @throws {Error} - If the API response is void or an error occurs while retrieving the country information.
+ */
+const getCountry = async (code) => {
+  try {
+    const response = await externalApi.countries.getCountry(code);
+
+    if (!response) {
+      throw new Error('Void response');
+    }
+
+    return response;
+  } catch (error) {
+    console.error('Unable to get country %o', error);
+    throw new Error('Unable to get country', { cause: error });
+  }
 };
 
-exports.findOneCountry = findOneCountry;
+/**
+ * Function to find one country.
+ *
+ * @param {Object} req - The request object.
+ * @param {Object} res - The response object.
+ * @returns {Object} - A promise that resolves with the response data.
+ * @throws {Error} - If unable to find one country.
+ */
+const findOne = async (req, res) => {
+  try {
+    const response = await getCountry(req.params.code);
 
-exports.findAll = async (req, res) => {
-  const countries = await externalApi.countries.getCountries();
-  const sortedCountries = sortCountries(countries);
-  res.status(200).send({
-    count: sortedCountries.length,
-    countries: sortedCountries,
-  });
+    if (!response) {
+      throw new Error('Void response');
+    }
+
+    const { status, data } = response;
+
+    if (data) {
+      return res.status(status).send(data);
+    }
+
+    return res.status(status).send({});
+  } catch (error) {
+    console.error('Unable to find one country %o', error);
+    throw new Error('Unable to find one country', { cause: error });
```

Comment:
Considering this is a controller do we want throw errors here or do we want to handle them, returning the response (ala `sign-in-link.controller.js`)

Reply:
Have updated.